cncserver.js

```diff
@@ -1979,6 +1979,42 @@ function simulationModeInit() {
   cncserver.pen.simulation = 1;
 }
 
+
+/**
+ * Helper function to implement matching port information to configured bot parameters.
+ *
+ * Attempts to match serialport.list output to parameters and set global
+ * 'serialPath' to matching port.
+ *
+ * @param {object} portDesc
+ *   Single port description returned by serialport.list
+ * @param {object} botControllerConf
+ *   The configured bot controller to try to match
+ */
+function autoDetectPort(portDesc, botControllerConf) {
```

Comment:
Should we export this for direct node module use? Seems like a handy thing to have. Also could remove the `portDesc` and have the listing generation be self contained in the function. See comment below on 2048 for more on refactoring this a little more.